ARE/components/actuator.knx/src/main/java/eu/asterics/component/actuator/knx/KnxInstance.java

```diff
@@ -696,42 +697,84 @@ public void receiveData(byte[] data) {
         @Override
         public void receiveData(byte[] data) {
             in1 = ConversionUtils.doubleFromBytes(data);
-            sendKNX(propGroupAddressSlider1, propertyToDPTid(propDPTSlider1), Double.toString(in1));
-        }
+            String DPTid = propertyToDPTid(propDPTSlider1);
+            String DPTval = null;
+            if(DPTDataTypeIsInt(DPTid)){

```

Comment:
Lines 701-706 should be done in one line like: `String DPTval = convertKNXValue(in1, DPTid)` with an additional helper method. Also in all other places where you do this to avoid duplicated code.